src/os.cpp

```diff
@@ -989,12 +989,28 @@ int os_self_exe_path(Buf *out_path) {
     }
 
 #elif defined(ZIG_OS_DARWIN)
+    // How long is the executable's path?
     uint32_t u32_len = 0;
     int ret1 = _NSGetExecutablePath(nullptr, &u32_len);
     assert(ret1 != 0);
-    buf_resize(out_path, u32_len);
-    int ret2 = _NSGetExecutablePath(buf_ptr(out_path), &u32_len);
+
+    // Allocate a buffer for this path.
+    Buf *path_tmp = buf_alloc_fixed(u32_len);
+    // Fill the buffer with the path, which may be a symlink.
+    int ret2 = _NSGetExecutablePath(buf_ptr(path_tmp), &u32_len);
     assert(ret2 == 0);
+
+    // Make a buffer with room for the real path.
+    Buf *resolve_tmp = buf_alloc_fixed(PATH_MAX);
+
+    // Fill it with the real resolved path.
+    char *real_path = realpath(buf_ptr(path_tmp), buf_ptr(resolve_tmp));
```

Comment:
Instead of `resolve_tmp`, we should resize `out_path` and use it directly. And then we need to `buf_resize(out_path, strlen(buf_ptr(out_path)))` to make the length correct.

Can you confirm this patch works for you?

Reply:
Works for me.